.github/workflows/ci_cd.yml

```
@@ -22,7 +22,7 @@ jobs:
             ~/.cargo/registry/cache
             ~/.cargo/registry/index
             core/target
-          key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}
+          key: cargo-debug-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}
```

I would put the profile at the end and enable partial cache hits so that at least something is shared between runs. Or we can separate core/target to another cache but share cache for the cargo registry.

```suggestion
          key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}-debug
          restore-keys: |
            cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}-
```